server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -45,8 +46,8 @@
      * it replaces "org.elasticsearch" with "org.elasticsearch.deprecation" to maintain
      * the "org.elasticsearch" namespace.
      */
-    private DeprecationLogger(Logger parentLogger) {
-        deprecationLogger = new ThrottlingAndHeaderWarningLogger(parentLogger);
+    public DeprecationLogger(Logger parentLogger) {
+        deprecationLogger = new ThrottlingAndHeaderWarningLogger(deprecatedLoggerName(parentLogger.getName()), indexingService);
```

Comment:
Do I need to be more defensive about how `indexingService` is set and supplied to the logger here?

Reply:
some assertion might indeed be helpful. It would prevent some accidental refactoring failures - people might not notice until some deprecation log is actually triggered.
I would also expect soem deprecation logger test to fail now.